File: ClientApp/src/dtos/TaskDto.js
export class TaskDto {
    constructor(id, title, description, dueDate, priority, isRecurring, isComplete, userId, tags){
        this.id = id || null;
        this.title = title;
        this.description = description;
        this.dueDate = dueDate;
        this.priority = priority;
        this.isRecurring = isRecurring;
        this.isComplete = isComplete;
        this.userId = userId;
        this.tags = tags || null;
    }
}

File: Dtos/TaskDto.cs
using System;
using System.Collections.Generic;
using Productivity.Enums;
using Productivity.Models;

namespace Productivity.Dtos;

public class TaskDto
{
    public long? Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime DueDate { get; set; }
    public Priority Priority { get; set; }
    public bool IsRecurring { get; set; }
    public bool IsComplete { get; set; }
    public string UserId { get; set; }
    public List<Tag>? Tags { get; set; }
}